src/components/panel/panel.js

```diff
@@ -961,37 +929,29 @@ MdPanelRef.prototype.detach = function() {
  */
 MdPanelRef.prototype.show = function() {
   if (!this._panelContainer) {
-    return this._$q.reject(
-        'Panel does not exist yet. Call open() or attach().');
+    return this._$q(function(resolve, reject) {
+      reject('Panel does not exist yet. Call open() or attach().');
+    });
   }
 
-  if (!this._panelContainer.hasClass(MD_PANEL_HIDDEN) && this._showPromise) {
-    return this._showPromise;
+  if (!this._panelContainer.hasClass(MD_PANEL_HIDDEN)) {
+    return this._$q.when(this);
   }
 
-  // TODO(ErinCoughlan) - Cancel any in-progress actions.
-
   var self = this;
-
   var animatePromise = function() {
     self.removeClass(MD_PANEL_HIDDEN);
     return self._animateOpen();
   };
 
-  this._showPromise = this._$q.all([
-      this._backdropRef ? this._backdropRef.show() : this._$q.resolve(self),
-      animatePromise().then(function() {
-        return self._$q.resolve(self);
-      }, function() {
-        self._$log.warn('MdPanel Animations failed. Showing panel without animating.');
-        return self._$q.resolve(self);
-      }).then(function() {
-        self._focusOnOpen();
-        return self._$q.resolve(self);
-      }, self._$q.reject)
-  ]);
-
-  return this._showPromise;
+  return this._$q(function(resolve, reject) {
+    var done = self._done(resolve, self);
+
+    self._$q.all([
+      self._backdropRef ? self._backdropRef.show() : self,
+      animatePromise().then(function() { self._focusOnOpen(); })
```

Comment:
Doesn't this `then` also need a reject? (I think this will break in Angular 1.6)